flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -599,20 +608,117 @@ public ExecutionGraph getExecutionGraph() {
     @Override
     public final void startScheduling() {
         mainThreadExecutor.assertRunningInMainThread();
-        registerJobMetrics(jobManagerJobMetricGroup, executionGraph, this::getNumberOfRestarts);
+        registerJobMetrics(
+                jobManagerJobMetricGroup,
+                executionGraph,
+                this::getNumberOfRestarts,
+                ignored -> {},
+                executionGraph.getStatusTimestamp(JobStatus.INITIALIZING),
+                jobStatusMetricsSettings);
         operatorCoordinatorHandler.startAllOperatorCoordinators();
         startSchedulingInternal();
     }
 
     public static void registerJobMetrics(
```

Comment:
Would it make sense to get rid of the `jobStatusListenerConsumer`? I've found it bit confusing at first.
```suggestion
    public static JobStatusListener registerJobStatusMetrics(
```

Reply:
I'm not too fond of it either, but returning something doesn't scale in the long-term imo. What happens the next time we'd need to add a parameter that isn't quite as nice as we'd like?